react-components/src/utils/createProductPageUrl.js

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ *
+ *    Copyright 2021 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+export const createProductPageUrl = sku => {
+    const url = new URL(window.location);
+
+    // Provided by StoreConfigExporter
+    const pathname = document.body.dataset.storeRootUrl;
+    if (!pathname) {
+        return null;
+    }
+
+    const extension = pathname.substr(pathname.lastIndexOf('.'));
```

Comment:
We should fix the extension to `html` as this is also how the servlet is registered.